DHI inline patch (cassandra 4.0): drop unused Hadoop transitive jars from runtime lib

Excludes vulnerable transitive dependencies pulled in by provided-scope
hadoop-core and hadoop-minicluster so they are not copied into
usr/share/cassandra/lib: commons-collections, commons-beanutils, Jasper,
Mina, Jersey, commons-net, commons-lang (CVE-2025-48924), commons-httpclient
(CVE-2012-5783), and commons-configuration (CVE-2025-46392). None of those
code paths are exercised by core Cassandra (CQL, gossip, nodetool).

Patch context: https://github.com/apache/cassandra/blob/cassandra-4.0.21/build.xml
(applies with fuzz 1 to the 4.0.21 tag used by package/main/cassandra/debian-13/4.0.yaml)
Build wiring: package/main/cassandra/patch/cassandra-4-mvn-dependencies.json

diff --git a/build.xml b/build.xml
index 0d7a567a91..72ed21d024 100644
--- a/build.xml
+++ b/build.xml
@@ -598,11 +598,24 @@
             <exclusion groupId="ant" artifactId="ant"/>
             <exclusion groupId="junit" artifactId="junit"/>
             <exclusion groupId="org.codehaus.jackson" artifactId="jackson-mapper-asl"/>
+            <exclusion groupId="commons-collections" artifactId="commons-collections"/>
+            <exclusion groupId="commons-beanutils" artifactId="commons-beanutils"/>
+            <exclusion groupId="tomcat" artifactId="jasper-runtime"/>
+            <exclusion groupId="tomcat" artifactId="jasper-compiler"/>
+            <exclusion groupId="com.sun.jersey" artifactId="jersey-server"/>
+            <exclusion groupId="com.sun.jersey" artifactId="jersey-core"/>
+            <exclusion groupId="commons-net" artifactId="commons-net"/>
+            <exclusion groupId="commons-lang" artifactId="commons-lang"/>
+            <exclusion groupId="commons-httpclient" artifactId="commons-httpclient"/>
+            <exclusion groupId="commons-configuration" artifactId="commons-configuration"/>
             <exclusion groupId="org.slf4j" artifactId="slf4j-api"/>
           </dependency>
           <dependency groupId="org.apache.hadoop" artifactId="hadoop-minicluster" version="1.0.3" scope="provided">
             <exclusion groupId="asm" artifactId="asm"/> <!-- this is the outdated version 3.1 -->
             <exclusion groupId="org.slf4j" artifactId="slf4j-api"/>
             <exclusion groupId="org.codehaus.jackson" artifactId="jackson-mapper-asl"/>
+            <exclusion groupId="org.apache.hadoop" artifactId="hadoop-test"/>
+            <exclusion groupId="com.sun.jersey" artifactId="jersey-server"/>
+            <exclusion groupId="com.sun.jersey" artifactId="jersey-core"/>
           </dependency>
           <dependency groupId="net.java.dev.jna" artifactId="jna" version="5.6.0"/>
